Library/Homebrew/cli/named_args.rb

```diff
@@ -29,13 +42,212 @@ def initialize(*args, parent: Args.new, override_spec: nil, force_bottle: false,
+
+        paths += converted_args[:formula].map(&:path)
+        paths += converted_args[:cask].map(&:sourcefile_path)
+        paths += converted_args[:tap].map(&:path)
+
+        paths
+      end
+
+      private
+
+      sig { params(installed: T::Boolean, ignore_unavailable: T::Boolean).returns(T::Array[Formula]) }
+      def formulae(installed: false, ignore_unavailable: false)
+        downcased_unique_named.flat_map do |name|
+          formula = Formulary.resolve(name, *spec, force_bottle: @force_bottle, flags: @flags)
+
+          raise FormulaNotInstalledError, name if installed && !formula.any_version_installed?
+
+          @args_to_convert.delete_at @args_to_convert.index(name) if @args_to_convert.include? name
+          formula
+        rescue FormulaUnavailableError, FormulaNotInstalledError
+          raise unless ignore_unavailable
+        end.compact.uniq.freeze
+      end
+
+      sig { params(installed: T::Boolean, ignore_unavailable: T::Boolean).returns(T::Array[Cask::Cask]) }
+      def casks(installed: false, ignore_unavailable: false)
+        downcased_unique_named.flat_map do |name|
+          cask = begin
+            Cask::CaskLoader.load(name, config: Cask::Config.from_args(@parent))
+          rescue Cask::CaskUnavailableError
+            retry if Tap.install_default_cask_tap_if_necessary
+            raise
+          end
+
+          raise Cask::CaskNotInstalledError, name if installed && !cask.installed?
+
+          @args_to_convert.delete_at @args_to_convert.index(name) if @args_to_convert.include? name
+          cask
+        rescue Cask::CaskUnavailableError, Cask::CaskNotInstalledError
+          raise unless ignore_unavailable
+        end.compact.uniq.freeze
+      end
+
+      sig { params(installed: T::Boolean, ignore_unavailable: T::Boolean).returns(T::Array[Tap]) }
+      def taps(installed: false, ignore_unavailable: false)
+        downcased_unique_named.flat_map do |name|
+          tap = Tap.fetch(name)
+
+          raise TapUnavailableError, name if installed && !tap.installed?
+
+          @args_to_convert.delete_at @args_to_convert.index(name) if @args_to_convert.include? name
+          tap
+        rescue RuntimeError, TapUnavailableError
+          raise unless ignore_unavailable
+        end.compact.uniq.freeze
+      end
+
+      sig { returns(T::Array[String]) }
+      def downcased_unique_named
+        # Only lowercase names, not paths, bottle filenames or URLs
+        map do |arg|
+          if arg.include?("/") || arg.end_with?(".tar.gz") || File.exist?(arg)
+            arg
+          else
+            arg.downcase
+          end
+        end.uniq
+      end
+
+      def spec
+        @override_spec
+      end
+
+      # TODO: (2.10) Remove the rest of this class once the remaining methods are removed
+      public
+
       def to_casks
+        # TODO: (2.8) Deprecate this method
+        # odeprecated "`args.named.to_casks`", "`args.named.convert` and `named_args :cask`"
```

Comment:
I think these methods (and those below) can be considered private and we can just adjust them ourselves. I've never seen any tap usage and they aren't documented.

Reply:
If that's the case, it makes my life much easier 😅!